### ISSUE TITLE
0.4.0 Release

### ISSUE THREAD

**Issue by Ehesp** [react-native-material-design/react-native-material-design#12]
> This list is a work in progress 

- [ ] Refactor List component

**Comment by bozzmob** [comment]
Anything more planned for 0.4.0? I guess you are handling the List Refactor.

**Comment by Ehesp** [comment]
The GridComponent? Also I need to debug a Ripple bug where the state persists. Any other ideas I'm open to!

**Comment by dgilperez** [comment]
What about a FAB?

**Comment by Ehesp** [comment]
Ah yeah. How should the position of it be done? Simply render the FAB and
let the user decide where to place it?

**Comment by dgilperez** [comment]
Hmm not really sure. Specs about fabs say how they should be positioned
with relation to other objects, and they seem to be always bottom-right ...
Maybe we can style them lime that by default and let the user override?

**Comment by pewh** [comment]
What about a Bottom Sheet?

**Comment by bozzmob** [comment]
Yes. The FAB has to be at the bottom-right according to the [spec](https://www.google.co.in/design/spec/components/buttons-floating-action-button.html#).
Though FAB comes in handy, but, personally I somewhat like this article - [Why the Floating Action Button is bad UX design](https://medium.com/tech-in-asia/material-design-why-the-floating-action-button-is-bad-ux-design-acd5b32c5ef#.efez2ak4q)

@Ehesp I can have a look at FAB, Gridview.

**Comment by bozzmob** [comment]
With respect to Grid View, I will be implementing this first (from the [spec](https://www.google.co.in/design/spec/components/grid-lists.html#))-

![image](https://material-design.storage.googleapis.com/publish/material_v_4/material_ext_publish/0Bx4BSt6jniD7YTU2ekJPMU80MVU/components_grids_usage1.png)

**Comment by Ehesp** [comment]
@pewh problem with a bottom sheet is that currently there's no way to force a component to the top of everything else (I believe).

**Comment by bozzmob** [comment]
@pewh @Ehesp There is a way to do that. But, not sure its available on Android yet. We can tweak concepts and code from the Modal and implement it I believe. I have seen such an implementation somewhere. Not sure how good or feasible it is.

**Comment by Ehesp** [comment]
@bozzmob Okay cool - Shall look into it.

What does everyone think of 'Swipe Views with Tabs'? I've got it working on another project using the project mentioned in the list, but had to create a custom tab bar to make it feel native...?

**Comment by Salakar** [comment]
![jan 06 2016 19 04](https://cloud.githubusercontent.com/assets/5347038/12151748/957020e0-b4a8-11e5-8523-8f734fc99873.gif)

Badge animations done!

**Comment by bozzmob** [comment]
@Ehesp Yeah! There is a ViewPager component which exists officially. We can make use of react-native-scrollable-tab-view as you have suggested. But, some css changes are needed so that the look and feel is similar to the existing components.

**Comment by Ehesp** [comment]
Nice work @Salakar, really cool.

@bozzmob I'll do that then as I've already got it on another project. `react-native-scrollable-tab-view` doesn't allow a single tab too, but I've got a way around that.

**Comment by bozzmob** [comment]
@Ehesp How about we use [react-native-action-button](https://github.com/mastermoo/react-native-action-button.git) for FAB?

**Comment by Ehesp** [comment]
That looks good! Will look if there's any benefit of pulling it in and making a wrapper around it. If not, it might just be best to suggest this for a FAB.

**Comment by zoontek** [comment]
A [dropdown menu](https://www.google.com/design/spec/components/menus.html) component could be awesome!

**Comment by Ehesp** [comment]
@zoontek I agree, however part of the problem we have (Bottom Sheets has this issue too), is that React Native does not yet support an "on the top of everything" API. So the dropdown menu might get stuck behind a component which is nested deeper down the chain.

**Comment by bozzmob** [comment]
I am working on the Gridview component now. Have made some progress. Still a lot to do.

**Comment by bozzmob** [comment]
**Screenshots of Gridview-**

Currently, its working for a fixed height, width.

I am working on making it a generic component. Currently most values like margin, padding are being hard-coded. So, thought of making it responsive.

![image](https://github.com/bozzmob/React-Native-Material-GridView/blob/master/screenshots/Screenshot_2016-01-08-00-46-23.png?raw=true)

![image](https://github.com/bozzmob/React-Native-Material-GridView/blob/master/screenshots/Screenshot_2016-01-08-00-06-42.png?raw=true)

**Comment by Salakar** [comment]
@bozzmob any updates on the grid view?

**Comment by bozzmob** [comment]
@Salakar Yes. I am working on it. It should be available in a couple of days.

**Comment by zoontek** [comment]
The dropdown is available in React-Native 0.19-rc: https://github.com/facebook/react-native/commit/18437093f2664beedd3239bb525d7daa3f2bcdbd :+1:

**Comment by Ehesp** [comment]
Awesome :D 

I just wish there was some "z-index" type style so we could push whatever we wanted to the top of the stack.

**Comment by zoontek** [comment]
@Ehesp Yes, or a custom component "Above"

```jsx
<Above
  backgroundVisible={true}
  backgroundColor="#000"
  backgroundOpacity={0.5}
>
  <View>
    <Text>Could be really usefull!</Text>
  </View>
</Above>
```

**Comment by zoontek** [comment]
Is the project dead? :(

**Comment by Ehesp** [comment]
Nope, was using it in a internal work project however that's not currently being developed so focus has shifted. Plan on picking it up in the next few weeks hopefully.

**Comment by zoontek** [comment]
Great! I will check if I can do the Swipe Views with Tabs!

**Comment by anhldbk** [comment]
@Ehesp Any chance to integrate with https://github.com/xinthink/react-native-material-kit?

**Comment by kkgelu** [comment]
I would like to have a proper dialog (in future releases), example: 
https://github.com/aakashns/react-native-dialogs
